### ISSUE TITLE
Универсальность решения

### ISSUE THREAD

**Issue by EugenioAV** [dmfilimonova/classwork#2]
**Для данных, которые были представлены в формулировки задачи все решено верно.**

Однако, при решении задачи также необходимо думать о том, что данные могут обновиться, добавиться новые значения в поле  'favoriteFruit'. Сейчас программный код будет работать только со значениями поля 'banana', 'strawberry', 'apple', так как на каждое отдельное слово у тебя есть проверка:
https://github.com/dmfilimonova/classwork/blob/6d647b462dbee4c95fa4a1cdc151a6d1bae84caf/work/1.1.py#L9

Если добавится какое-то новое значение в поле 'favoriteFruit', то нужно будет изменить код и создавать новую переменную для следующего списка. А если добавится 100 новых значений, то 100 новых списков. Поэтому, такое решение задачи не является универсальным.

Для универсальности решения, как вариант, можешь создать дополнительный словарь (dict), в который записывать новое значение, которое встретилось в этом поле. Таким образом будет два варианта:
- если в дополнительном словаре уже есть ключ, то просто добавляешь новое имя в список к значению по этому ключу.
- если в дополнительном словаре такого ключа нет, то создаешь ключ в этом словаре, и значение к нему в виде списка с именем человека.

Таким образом должна появится примерно такая структура, которую можно использовать в выводе ответа: 
`
{
    'apple': [...перечисление имен...],
    'banana': [...перечисление имен...],
    ....
}
`

**Comment by dmfilimonova** [comment]
Да, я рассматривала вариант решения через создание dict (), но возникли трудности с циклом, не понимала, как правильно реализовать задание цикла. 
Я ещё подумаю над этой задачей, но если не получится, можно будет попросить о помощи в четверг?



Отправлено из мобильной Почты Mail.ru


понедельник, 17 февраля 2020 г., 12:36 +0300 от notifications@github.com  <notifications@github.com>:

**Comment by EugenioAV** [comment]
Да, конечно!
